chamge/amqp-arbiter-backend.c

```diff
@@ -38,59 +44,37 @@ chamge_amqp_arbiter_backend_enroll (ChamgeArbiterBackend * arbiter_backend)
   g_autofree gchar *amqp_enroll_q_name = NULL;
   g_autofree gchar *amqp_exchange_name = NULL;
   gint amqp_channel;
-
-  struct amqp_connection_info amqp_c_info;
-  amqp_bytes_t queue;
-  amqp_queue_declare_ok_t *amqp_r = NULL;
+  GValue value_str = G_VALUE_INIT;
+  GValue value_int = G_VALUE_INIT;
 
   ChamgeAmqpArbiterBackend *self =
       CHAMGE_AMQP_ARBITER_BACKEND (arbiter_backend);
 
+  /* get configuration from gsetting */
   amqp_uri = g_settings_get_string (self->settings, "amqp-uri");
   g_assert_nonnull (amqp_uri);
-
   amqp_channel = g_settings_get_int (self->settings, "amqp-channel");
-
-  amqp_parse_url (amqp_uri, &amqp_c_info);
-
-  g_debug ("parsed amqp uri (host: %s, vhost: %s)", amqp_c_info.host,
-      amqp_c_info.vhost);
-
-  amqp_socket_open (self->amqp_socket, amqp_c_info.host, amqp_c_info.port);
-
-  /*
-   * TODO: The authentication method should be EXTERNAL
-   * if we don't want to share user/passwd
-   */
-  amqp_login (self->amqp_conn, amqp_c_info.vhost, 0, 131072, 0,
-      AMQP_SASL_METHOD_PLAIN, amqp_c_info.user, amqp_c_info.password);
-  amqp_channel_open (self->amqp_conn, amqp_channel);
-  amqp_get_rpc_reply (self->amqp_conn);
-
   amqp_enroll_q_name =
       g_settings_get_string (self->settings, "enroll-queue-name");
-  queue = amqp_cstring_bytes (amqp_enroll_q_name);
-  amqp_r =
-      amqp_queue_declare (self->amqp_conn, amqp_channel, queue, 0, 0, 0, 1,
-      amqp_empty_table);
-  amqp_get_rpc_reply (self->amqp_conn);
-
-  g_debug ("declaring a queue : %s", amqp_r->queue.bytes);
-
   amqp_exchange_name =
       g_settings_get_string (self->settings, "enroll-exchange-name");
 
-  amqp_queue_bind (self->amqp_conn, amqp_channel, amqp_r->queue,
-      amqp_cstring_bytes (amqp_exchange_name), queue, amqp_empty_table);
+  /* set amqp login information */
+  g_value_init (&value_str, G_TYPE_STRING);
+  g_value_init (&value_int, G_TYPE_UINT);
 
-  g_debug ("binding a queue (exchange: %s, bind_key: %s)", amqp_exchange_name,
-      amqp_enroll_q_name);
+  g_value_set_string (&value_str, amqp_uri);
+  g_object_set_property (G_OBJECT (self->amqp_rpc), "url", &value_str);
```

Comment:
`g_object_set()`  doesn’t need `GValue`.

you might want to use `g_object_set (G_OBJECT (self->amqp_rpc), “url”, amqp_uri, NULL)`.

Reply:
Yes. Thanks!!